superset-frontend/src/explore/store.js

```diff
@@ -64,8 +64,14 @@ export function getControlsState(state, inputFormData) {
 export function applyDefaultFormData(inputFormData) {
   const datasourceType = inputFormData.datasource.split('__')[1];
   const vizType = inputFormData.viz_type;
-  const controlsState = getAllControlsState(vizType, datasourceType, null, {
-    ...inputFormData,
+  let controlsState = getAllControlsState(vizType, datasourceType, {
+    form_data: inputFormData,
+  });
```

Comment:
Here I'm removing the shallow copy of `inputFormData` and placing it in the `Partial<ControlPanelState>` object, as I didn't see any downstream code mutating the object.